java/vector/src/test/java/org/apache/arrow/vector/util/TestVectorAppender.java

```diff
@@ -120,6 +142,26 @@ public void testAppendVariableWidthVector() {
     }
   }
 
+  @Test
+  public void testAppendEmptyVariableWidthVector() {
+    try (VarCharVector target = new VarCharVector("", allocator);
+         VarCharVector delta = new VarCharVector("", allocator)) {
+
+      target.allocateNew(5, 10);
```

Comment:
This is not needed either, as `ValueVectorDataPopulator.setVector` calls the `getSafe` method.